drivers/clock_control: stm32h7: expanded with PLL2 and MCO support

SpooniSpoon opened the pull request:
drivers/clock_control: stm32h7: expanded with PLL2 and MCO support

Clock control driver for stm32h7 expanded with kconfig definitions
for PLL2 and additional MCO source selections and the related source
code modifications.

Signed-off-by: Raphael Löffel <loeffel@rte-ag.ch>

FRASTM commented on drivers/clock_control/Kconfig.stm32:
```diff
@@ -252,6 +252,18 @@ config CLOCK_STM32_MCO1_SRC_PLLCLK
+	  Use RC48 as source of MCO2
+
+config CLOCK_STM32_MCO2_SRC_PLL1Q
+	bool "PLL1Q"
```
Could it be  'config CLOCK_STM32_MCO**2**_SRC_PLL1**P**' ? 
and  ' config CLOCK_STM32_MCO**1**_SRC_PLL**1Q**'

As from the ref man (RM0468) , MCO1 /MCO2 clock sources in the RCC clock configuration register (RCC_CFGR) are: 

**MCO1**[2:0]: Micro-controller clock output 1
```
000: HSI clock selected (hsi_ck) (default after reset)
001: LSE oscillator clock selected (lse_ck)
010: HSE clock selected (hse_ck)
011: PLL1 clock selected (pll1_q_ck)
100: HSI48 clock selected (hsi48_ck)

```
**MCO2**[2:0]: Micro-controller clock output 2
```
000: System clock selected (sys_ck) (default after reset)
001: PLL2 oscillator clock selected (pll2_p_ck)
010: HSE clock selected (hse_ck)
011: PLL1 clock selected (pll1_p_ck)
100: CSI clock selected (csi_ck)
101:LSI clock selected (lsi_ck)
```

erwango reviewed (changes requested):
It is not possible to introduce Kconfig symbols that are not used in the code.
Adding PLL2 support is welcome but code that use it should also be provided.

Also, please consider https://github.com/zephyrproject-rtos/zephyr/pull/34701 in which we are transitiononing use of Kconfig by device tree for clocks configuration.

Also, please take care to provide valid commit messages (with short description and signed-off-by... (cf any commit you can find in tree)

SpooniSpoon commented:
Support of PPL2 is already part of the code. Only the kconfig symbols are missing. I use PPL2 to configure MCO2 as the 25MHz clock source for our ethernet phy.

erwango commented:
> Support of PPL2 is already part of the code. 

I see some PLL2 support for other series but not in H7, did I missed something ?

erwango reviewed (changes requested):
Thanks for this contribution.

We're transitioning clock control configuration from Kconfig to device tree on whole stm32 series. Change for H7 is about to be merged: https://github.com/zephyrproject-rtos/zephyr/pull/34701#pullrequestreview-648926401.
Once this transition is complete (target is DV2.6.0, ie next week), Kconfig related symbols will be deprecated.

So instead of adding new Kconfig symbols that would be deprecated immediately, can you convert your development to the use of new device tree configuration scheme ?
This is required at least for the PLL2 part of this change.
For the MCO part, conversion to device tree has not been done yet, but it would be the logical next step. So ideally, I'd like H7 MCO to use device tree as well. But since there is no equivalent in tree for now, this part is not blocking.

SpooniSpoon commented:
> Thanks for this contribution.
> 
> We're transitioning clock control configuration from Kconfig to device tree on whole stm32 series. Change for H7 is about to be merged: [#34701 (review)](https://github.com/zephyrproject-rtos/zephyr/pull/34701#pullrequestreview-648926401).
> Once this transition is complete (target is DV2.6.0, ie next week), Kconfig related symbols will be deprecated.
> 
> So instead of adding new Kconfig symbols that would be deprecated immediately, can you convert your development to the use of new device tree configuration scheme ?
> This is required at least for the PLL2 part of this change.
> For the MCO part, conversion to device tree has not been done yet, but it would be the logical next step. So ideally, I'd like H7 MCO to use device tree as well. But since there is no equivalent in tree for now, this part is not blocking.

Thanks for explanation. We will switch to V2.6.0 as soon as it released and we will convert our application to the device tree configuration scheme.

erwango commented:
> We will switch to V2.6.0 as soon as it is released and we will convert our application to the device tree configuration scheme.

Thanks for your understanding!

SpooniSpoon committed:
drivers/clock_control: stm32h7: expanded with PLL2 and MCO support

Clock control driver for stm32h7 expanded with kconfig definitions
for PLL2 and additional MCO source selections and the related source
code modifications.

Signed-off-by: Raphael Löffel <loeffel@rte-ag.ch>
